model_analyzer/config/generate/quick_run_config_generator.py

```diff
@@ -280,26 +282,105 @@ def _get_coordinate_values(self, coordinate: Coordinate,
     def _get_next_run_config(self) -> RunConfig:
         run_config = RunConfig(self._triton_env)
 
-        for i, _ in enumerate(self._models):
-            mrc = self._get_next_model_run_config(i)
+        model_index = 0
+        for model in self._models:
+            mrc, model_index = self._get_next_model_run_config(
+                model, model_index)
             run_config.add_model_run_config(mrc)
 
         return run_config
 
-    def _get_next_model_run_config(self, model_num: int) -> ModelRunConfig:
-        mc = self._get_next_model_config(model_num)
+    def _get_next_model_run_config(
+            self, model: ModelProfileSpec,
+            model_index: int) -> Tuple[ModelRunConfig, int]:
+        start_model_index = deepcopy(model_index)
+
+        if model.model_name() in self._ensemble_submodels:
+            ensemble_subconfigs, end_model_index = self._get_next_ensemble_subconfigs(
+                model, start_model_index)
+
+            param_combo = self._get_next_ensemble_param_combo(
+                start_model_index, end_model_index)
+
+            model_config = self._get_next_ensemble_model_config(
+                model, ensemble_subconfigs, param_combo)
+
+            model_index = end_model_index
+        else:
+            model_config = self._get_next_model_config(model, model_index)
+            model_index = model_index + 1
+
+        model_variant_name = model_config.get_field('name')
+        perf_analyzer_config = self._get_next_perf_analyzer_config(
+            model_variant_name, model, start_model_index)
+
+        model_name = model.model_name()
+        model_run_config = ModelRunConfig(model_name, model_config,
+                                          perf_analyzer_config)
+
+        if model.model_name() in self._ensemble_submodels:
+            model_run_config.add_ensemble_submodel_configs(ensemble_subconfigs)
+
+        return (model_run_config, model_index)
+
+    def _get_next_ensemble_subconfigs(
+            self, model: ModelProfileSpec,
+            start_model_index: int) -> Tuple[List[ModelConfig], int]:
+        model_index = deepcopy(start_model_index)
+        ensemble_subconfigs = []
+        for ensemble_submodel in self._ensemble_submodels[model.model_name()]:
+            ensemble_subconfigs.append(
+                self._get_next_model_config(ensemble_submodel, model_index))
+            model_index = model_index + 1
 
-        model_variant_name = mc.get_field('name')
-        pac = self._get_next_perf_analyzer_config(model_variant_name, model_num)
+        return (ensemble_subconfigs, model_index)
 
-        model_name = self._models[model_num].model_name()
-        return ModelRunConfig(model_name, mc, pac)
+    def _get_next_ensemble_param_combo(self, start_model_index: int,
+                                       end_model_index: int) -> dict:
+        """
+        For the ensemble model the only parameter we need to set 
```

Comment:
Have you confirmed this statement with Iman?

Reply:
Yes